### ISSUE TITLE
Can't finish setup

### ISSUE THREAD

**Issue by ArvsIndrarys** [poanetwork/poa-bridge-contracts#75]
I currently have problems finishing the setup :  
- I never was able to call the `initialize` method from the proxies
- On the Foreign bridge, the `initialize` method doesn't ask for daily limit, max_per_tx nor min_per_tx
- Trying to call them afterwards always fail

![pbc_dailylimit_failure](https://user-images.githubusercontent.com/19983929/44840833-4299e100-ac42-11e8-9e32-6dc5a3b13068.jpg)
![pbc_proxy_initialize_not_showing](https://user-images.githubusercontent.com/19983929/44840944-7c6ae780-ac42-11e8-8069-9c796a8d8e86.jpg)

I have currently tried the setup on the Foreign side only.

Branch used:  `refactor_v1`
Ethereum client : `Parity-Ethereum/v2.0.1-unstable-e7dc0be-20180726/x86_64-linux-gnu/rustc1.28.0`
Parity UI version : `0.3.4`

**Comment by ArvsIndrarys** [comment]
On an another point, the flatten script didn't add the `BridgeValidators` contract, I had to flatten it from the contract present in `contracts/upgradeable_contracts/BridgeValidators.sol`. Is it the right thing to do?

**Comment by akolotov** [comment]
@ArvsIndrarys
1. Are you trying to call `initialize` method directly, right? And you don't use the deployment script?

2. Which bridge mode are you trying to configure? If it is `erc-to-erc`, Foreign bridge contract has no ability to control erc20 token (ERC20 standard does not allow to notify contracts that a token ownership was transferred to them), that's why the bridge application listen `Transfer` events directly. Due to this the bridge contract does not monitor token transfer limits.

**Comment by ArvsIndrarys** [comment]
@akolotov 
1. Yes and yes, I am not using truffle, I am deploying on a private Parity network.
2. So the bridge should work event though the bounds are min=0 - max=0 ?

**Comment by akolotov** [comment]
@ArvsIndrarys `I am not using truffle, I am deploying on a private Parity network.` - so, did you study our deployment script and perform your deployment in similar manner? Was your actions automated? Which language are you using? I am asking to understand if we could study your script to find differences.

**Comment by akolotov** [comment]
`So the bridge should work event though the bounds are min=0 - max=0 ?` - that's applicable for Foreign side in case of `erc-to-erc` mode.

**Comment by ArvsIndrarys** [comment]
So I am trying to setup a bridge between two private networks : one at localhost:8545 and the other at localhost:8555.

As `NETWORK=sokol npm run deploy` takes only one parameter, and as I don't want to use kovan/mainnet for my tests, I decided to check in the deploy folder.

Given the README : 
0. compile source contracts:
I used the `compile.sh` script. 

1. create .env file : 
As that file seems to need addresses, I guessed I had to deploy the contracts beforehand. To do that, the indications seemed to be in the base README.

2. I deployed the contract given these indications -- the `BridgeValidators` was not appearing in the flats folder though, so I flattened it from what was in the contract folder.

I am curious about some things : 
- this is the list of calls on the Home bridge
```
claimTokens
executeAffirmation
initialize
onTokenTransfer
SetDailyLimit
SetGasPrice
SetMaxPerTx
SetMinPerTxSetRequiredBlockConfirmations
submitSignature
```
- and the list of calls on the Foreign bridge
```
claimTokens
executeSignature
initialize
SetDailyLimit
SetGasPrice
SetMaxPerTx
SetMinPerTx
SetRequiredBlockConfirmations
```

What would be the contract call to do on each side to transfer assets from one bridge to another?
I guess that from now on, if I want the contracts to work, I would need to setup the bridge from https://github.com/poanetwork/poa-bridge ?
